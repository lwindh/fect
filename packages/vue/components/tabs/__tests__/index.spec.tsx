import { mount } from '@vue/test-utils'

import Tabs from '../index'
import Tab from '../../tab'

describe('Tabs', () => {
  it('should be render as element', () => {
    const wrapper = mount(Tabs)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be change acitve value when tab trigger', async () => {
    const wrapper = await mount({
      render() {
        return (
          <Tabs v-model={[this.active, ['active']]}>
            <Tab title="HTML">HTML</Tab>
            <Tab title="JS">JS</Tab>
          </Tabs>
        )
      },
      data() {
        return {
          active: 0,
        }
      },
    })
    const els = wrapper.findAll('.fect-tabs__title') // filter all tabs title
    await els[1].trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be disabled all events', async () => {
    const wrapper = await mount({
      render() {
        return (
          <Tabs v-model={[this.active, ['active']]} hideDivider={false}>
            <Tab title="HTML" disabled={this.disabled}>
              HTML
            </Tab>
            <Tab title="JS">JS</Tab>
          </Tabs>
        )
      },
      data() {
        return {
          active: 0,
          disabled: true,
        }
      },
    })
    const els = wrapper.findAll('.fect-tabs__title') // filter all tabs title
    await els[0].trigger('click')
    await wrapper.setProps({
      hideDivider: true,
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Tab can not render without Tabs', () => {
    const wrapper = mount(Tab)
    expect(() => wrapper.get('div')).toThrowError()
  })
})
