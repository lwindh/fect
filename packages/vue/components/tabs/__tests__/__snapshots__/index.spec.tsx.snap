// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should be change acitve value when tab trigger 1`] = `
<div class="fect-tabs" modelvalue="0" modelmodifiers="[object Object]">
  <header class="fect-tabs__header ">
    <div role="button" class="fect-tabs__title ">HTML</div>
    <div role="button" class="fect-tabs__title active">JS</div>
  </header>
  <div class="fect-tab none">HTML</div>
  <div class="fect-tab ">JS</div>
</div>
`;

exports[`Tabs should be disabled all events 1`] = `
<div class="fect-tabs" modelvalue="0" modelmodifiers="[object Object]">
  <header class="fect-tabs__header fect-tabs__header--hide-divider">
    <div role="button" class="fect-tabs__title active disabled">HTML</div>
    <div role="button" class="fect-tabs__title ">JS</div>
  </header>
  <div class="fect-tab ">HTML</div>
  <div class="fect-tab none">JS</div>
</div>
`;
